src/main.c

```diff
@@ -369,21 +354,22 @@ int main(int argc, char **argv)
 
 	if (optind < argc - 1 || optind > argc)
 		goto user_error;
-	set_syslog(cfg.use_syslog);
 
 	if (password != NULL)
 		log_warn("You should not pass the password on the command line. Type it interactively or use a config file instead.\n");
 
 	// Load config file
 	if (config_file[0] != '\0') {
 		ret = load_config(&cfg, config_file);
-		set_syslog(cfg.use_syslog);
 		if (ret == 0)
 			log_debug("Loaded config file \"%s\".\n", config_file);
 		else
 			log_warn("Could not load config file \"%s\" (%s).\n",
 			         config_file, err_cfg_str(ret));
 	}
+	// Then apply CLI config
+	merge_config(&cfg, &cli_cfg);
+	set_syslog(cfg.use_syslog);
 
 	// Read host and port from the command line
 	if (optind == argc - 1) {
```

Comment:
!$#@ github for not being able to comment on arbitrary lines.. 20 or so lines below this I noticed some option parsings are redundant with the merge (e.g. reset username if it was given on command line etc) -- I'll update these to write in cli_cfg directly and let the merge function do the job